lib/Dialect/Torch/Transforms/InlineGlobalSlots.cpp

```diff
@@ -239,14 +239,15 @@ bool InlineGlobalSlotsAnalysis::isValueSafeTransferFunction(Value value) {
       continue;
     // If the op is read-only and all results are safe, then this value is
     // safe. This covers, for example, view-like ops that create aliases.
-    if ((op->hasTrait<Torch::OpTrait::ReadOnly>() ||
-         MemoryEffectOpInterface::hasNoEffect(op)) &&
-        llvm::all_of(op->getResults(), [&](Value result) {
-          auto *state =
-              getOrCreateFor<InlineGlobalSlotsAnalysisState>(value, result);
-          return state->isSafe;
-        }))
-      continue;
+    if (auto effects = dyn_cast<MemoryEffectOpInterface>(op)) {
```

Comment:
The torchdynamo tests are failing because this is not correct. The only change that needs to happen in this file is replacing `MemoryEffectOpInterface::hasNoEffect` with `isMemoryEffectFree`.

See: https://reviews.llvm.org/D137857

Reply:
Thanks, @ramiro050, I didn't think that this could be an issue.